romancal/source_detection/source_detection_step.py

```diff
@@ -122,8 +140,10 @@ def process(self, input):
                 sources = daofind(self.data, mask=self.coverage_mask)
 
             elif self.calc_threshold is not None:
-                # subtrack background from data if calculating abs. threshold
+                # subtract background from data if calculating abs. threshold
                 sources = daofind(self.data - bkg.background, mask=self.coverage_mask)
+            else:
+                sources = daofind(self.data, mask=self.coverage_mask)
```

Comment:
Staring a little at the threshold logic, I'm confused.  daofind doesn't do it natively, I think, but morally one wants to find all point sources that are more than N sigma above background, for N ~ 5.  That looks potentially like what 'calc_threshold' mode is doing, but there it looks like the background is both part of the threshold and subtracted from the data, which seems like doubly subtracting.  Meanwhile the mode without calc_threshold doesn't look like it has a notion of RMS and instead only has a notion of background, and the threshold is set exactly at the background?

I'm not sure there are any cases where we don't want to measure and remove a background, so I think I'd recommend just removing the calc_threshold = False branch, and then fixing the calc_threshold = True branch so that doesn't ~doubly subtract the background?

Sorry, I realize that this isn't new code, etc., but we should still fix it.

Reply:
I honestly did not understand what was happening in this block either. I added the `else` block that you've commented on here to make this PR work, but I agree that the logic needs revision.